.travis.yml

```diff
@@ -23,15 +20,21 @@ script:
     VERSION=${VERSION}
     MILESTONE=${MILESTONE}
     MVN_PROFILE=${MVN_PROFILE}
+    TRAVIS_BRANCH=${TRAVIS_BRANCH}
+    TRAVIS_REPO_SLUG=${TRAVIS_REPO_SLUG}
+    TRAVIS_PULL_REQUEST=${TRAVIS_PULL_REQUEST}
 
     EOF
+- ./mvnw install -P docs -q -U -DskipTests=true -Dmaven.test.redirectTestOutputToFile=true
+- '[ "${MVN_GOAL}" == "deploy" ] && ./docs/src/main/asciidoc/ghpages.sh || echo "Not updating docs"'
+script:
```

Comment:
nit: when changing things that seem unrelated, self-note in the PR what's going. it isn't a problem to change other things per-se, just distracting to figure out what if anything this has to do with the core change.